### PR TITLE
[SG-34723] Accessibility: Search results misc keyboard bugs

### PR DESCRIPTION
**Pull request by gitstart-sourcegraph**
## Audit type
Keyboard navigation

User journey audit issue
[#34190](https://github.com/sourcegraph/sourcegraph/issues/34190)

## Refs
[Gitstart ticket](https://app.gitstart.com/clients/sourcegraph/tickets/SG-34723)
[Sourcegraph issue](https://github.com/sourcegraph/sourcegraph/issues/34723)

### Problem description
The following problems exist for keyboard navigation on the result page:
### On the header:
header
- [x] Cannot focus (i) icon next to results count with keyboard to see number of repos
- [x] "Create context", "Create insight", "Monitor" and "Save Search" buttons cannot be activated with spacebar
- [x] Clicking on "save search" and then clicking "save query" on the dialog navigates to the save search page, the description box should be focused by default

### On search results items:
search results items
- [x] Last sync time icon cannot be focused with keyboard to see the last synced time

### On the no results page
results page
- [x] "Run search" link in examples cannot be focused with the keyboard

**Odd focus ring around no results page examples**
  alt
- [x] Should not be able to focus toggles on no results page examples

### Test Plan

Ensure integration test have accessibility audit and that those tests pass.
Use screen readers ([check here how](https://docs.sourcegraph.com/dev/background-information/web/accessibility/how-to-screen-reader)) to go through the changes and it should read as expected.

## App preview:

- [Web](https://sg-web-contractors-sg-34723.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-ddxmlxfnyh.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/wildcard/src/components/ButtonLink/ButtonLink.tsx [hunk 1]
```diff
@@ -83,35 +96,27 @@ export const ButtonLink = React.forwardRef((props, reference) => {
...
-            <Button
-                {...commonProps}
-                as={AnchorLink}
-                to=""
-                onClick={handleClick}
-                onAuxClick={handleClick}
-                role="button"
-                {...rest}
-            >
+            <Button {...commonProps} as={AnchorLink} to="" onClick={handleClick} onAuxClick={handleClick} {...rest}>
```
Comment (gitstart-sourcegraph) [inline client/wildcard/src/components/ButtonLink/ButtonLink.tsx:119]: @limitedmage @umpox 

In this PR
- I set `role=button` by default, so all `ButtonLink` usages can be read by screen reader with "button" word at the end.
- By default browser doesn't trigger page redirections when using `spacebar` on anchor tags, so we trigger `click` event manually.

Comment (gitstart-sourcegraph) [reply client/wildcard/src/components/ButtonLink/ButtonLink.tsx:119]: cc @valerybugakov

Comment (limitedmage) [reply client/wildcard/src/components/ButtonLink/ButtonLink.tsx:119]: Alternatively, we can keeps these links-that-look-like-buttons as just links (and the screen reader will read them as such, and will require "enter" key to trigger them), but I think it's odd for sighted keyboard-only users to see a button that can't be triggered with the spacebar. Is there anything in the WCAG spec that mentions this?

**Review by limitedmage — APPROVED** [review]

File: client/wildcard/src/components/ButtonLink/ButtonLink.tsx (hunk 1 above)
Comment (gitstart-sourcegraph) [reply client/wildcard/src/components/ButtonLink/ButtonLink.tsx]: Thanks @limitedmage Actually couldn't find any spec for it, so I agree that we should keep the native role

**Review by valerybugakov — APPROVED** [review]
The solution with a programmatic click trigger looks good to me. I didn't find any red flags in the spec regarding this. 

@umpox, @limitedmage, let me know if you see any issues.

**Pull request merged**